### PR TITLE
fix: only purge `constructor.prototype` keys

### LINKED ISSUE #24: Should parse constructor unless `prototype` child
### Environment

none	

### Reproduction

See issues below

### Describe the bug

Right now, every `constructor` key will be removed. Instead, only `{ "constructor": { "prototype": "<...>" } }` should be removed.

See https://github.com/fastify/secure-json-parse/blob/master/test/index.test.js#L228-L231

### Additional context

https://github.com/unjs/ofetch/issues/180

### Logs

_No response_

### PR DESCRIPTION
**Pull request by manniL**
Resolves #24, resolves https://github.com/unjs/ofetch/issues/180, resolves https://github.com/nuxt/framework/issues/9468

Would really like to wait for #23 and add some tests for it too

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -6,7 +6,18 @@ const suspectConstructorRx = /"(?:c|\\u0063)(?:o|\\u006[Ff])(?:n|\\u006[Ee])(?:s
...
 function jsonParseTransform (key: string, value: any): any {
-  if (key === "__proto__" || key === "constructor") {
+  if (key === "__proto__") {
+    return;
+  }
+  if (key === "constructor") {
+    if (typeof value !== "object") {
+      return value;
+    }
+    const hasPrototype = Object.prototype.hasOwnProperty.call(value, "prototype");
```
Comment (pi0) [inline src/index.ts:16]: Shall we simplify it to `value && (`"prototype" in value)`?

Comment (manniL) [reply src/index.ts]: Like that?

Comment (pi0) [reply src/index.ts]: Check last commit.

File: src/index.ts [hunk 2]
```diff
@@ -6,7 +6,11 @@ const suspectConstructorRx = /"(?:c|\\u0063)(?:o|\\u006[Ff])(?:n|\\u006[Ee])(?:s
 const JsonSigRx = /^["[{]|^-?\d[\d.]{0,14}$/;
 
 function jsonParseTransform (key: string, value: any): any {
-  if (key === "__proto__" || key === "constructor") {
+  if (key === "__proto__") {
+    return;
+  }
+  if (key === "constructor" && value && typeof value === "object" && ("prototype" in value)) {
```
Comment (pi0) [inline src/index.ts:12]: To be honest i'm still worried about performance overhead of this new check. `jsonParseTransform` was a second assurance method in top of regex. But anyway let's have it as hotfix and revise later

Comment (manniL) [reply src/index.ts:12]: But that should only bring an overhead for these specific keys, right? 🤔

Comment (pi0) [reply src/index.ts:12]: Yes indeed :)

**Pull request merged**